src/Avalonia.Animation/AnimationInstance`1.cs

```diff
@@ -130,84 +130,71 @@ private void DoPlayStates()
+            var initDelay = _initialDelay.Ticks * _speedRatioConv;
 
-             // Determine if the current iteration should have its normalized time inverted.
-            bool isCurIterReverse = _animationDirection == PlaybackDirection.Normal ? false :
-                                    _animationDirection == PlaybackDirection.Alternate ? (_currentIteration % 2 == 0) ? false : true :
-                                    _animationDirection == PlaybackDirection.AlternateReverse ? (_currentIteration % 2 == 0) ? true : false :
-                                    _animationDirection == PlaybackDirection.Reverse ? true : false;
-   
-            if (!_isLooping)
-            {
-                var totalTime = _iterationDelay ? _repeatCount * ( _duration.Ticks + _delay.Ticks) : _repeatCount * _duration.Ticks + _delay.Ticks;
-                if (time.Ticks >= totalTime)
-                {
-                    var easedTime = _easeFunc.Ease(isCurIterReverse ? 0.0 : 1.0);
-                    _lastInterpValue = _interpolator(easedTime, _neutralValue);
-                   
-                    DoComplete();
-                    return;
-                }
-            }
-            iterationTime = TimeSpan.FromTicks((long)(iterationTime.Ticks % iterationEndpoint.Ticks));
-        
-            if (delayEndpoint > TimeSpan.Zero & iterationTime < delayEndpoint)
+            if (indexTime > 0 & indexTime <= initDelay)
             {
                 DoDelay();
             }
             else
             {
-                // Offset the delay time            
-                iterationTime -= delayEndpoint;
-                iterationEndpoint -= delayEndpoint;
+                // Calculate timebases.
+                var iterationTime = iterDuration + iterDelay;
+                var opsTime = indexTime - initDelay;
+                var playbackTime = opsTime % iterationTime;
 
-                // Normalize time
-                var interpVal = (double)iterationTime.Ticks / iterationEndpoint.Ticks;
+                _currentIteration = (ulong)(opsTime / iterationTime);
 
-                if (isCurIterReverse)
-                    interpVal = 1 - interpVal;
+                // Stop animation when the current iteration is beyond the iteration count.
+                if ((_currentIteration + 1) > _iterationCount)
+                    DoComplete();
 
-                // Ease and interpolate
-                var easedTime = _easeFunc.Ease(interpVal);
-                _lastInterpValue = _interpolator(easedTime, _neutralValue);
+                if (playbackTime <= iterDuration)
+                {
+                    // Normalize time for interpolation.
+                    var normalizedTime = playbackTime / iterDuration;
+
+                    // Check if normalized time needs to be reversed.
+                    bool isCurIterReverse = _animationDirection == PlaybackDirection.Normal ? false :
```

Comment:
This makes me hard to understand

Reply:
I think making this a switch case would be less cleaner than this but i'll see what i can do.